### PR TITLE
mark sections normative; move algorithms

### LINKED ISSUE #300: Almost no sections are `class="normative"`. Why?
https://github.com/w3c/vc-jose-cose/blob/befdeae56c01fb9f790bca348ddb57f9ab5f59f3/index.html#L394

This section is one of *very* few marked explicitly as `normative`, though the introduction says that anything not marked `non-normative` *is* normative.

Why is *this* section explicitly marked `normative`? Why are the others _not_?

### PR DESCRIPTION
**Pull request by decentralgabe**
fix #300 

* mark sections as normative, informative
* move algorithms above non-normative section
* combined verification and validation sections under a common 'algorithms' section

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/vc-jose-cose/pull/302.html" title="Last updated on Sep 26, 2024, 4:44 PM UTC (c530a98)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/vc-jose-cose/302/e0e7d91...c530a98.html" title="Last updated on Sep 26, 2024, 4:44 PM UTC (c530a98)">Diff</a>

### CODE REVIEW

**Comment by selfissued** [comment]
Where can I view the rendered version of this PR?  The usual "Preview" link isn't in evidence.

**Comment by iherman** [comment]
There seems to be some problem with the background processing that I could not track down. I believe it may be with the "spec generator". That is probably the reason why preview does not appear either.

The (complicated) way around that I found is as follows:

- download the new branch onto the local clone of repository and ask git to make this your current branch
- two possibilities: 
    - if you run a local http server which has access to the repository, then view the index.html file in your browser via localhost
    - otherwise, go to https://labs.w3.org/spec-generator/ and upload the source code there

In theory, one could use the spec generator using https://htmlpreview.github.io/?https://github.com/w3c/vc-jose-cose/blob/fix-300/index.html to render the file, but it returns a (I believe unjustified) but report.

cc @deniak ?

**Comment by deniak** [comment]
As far as I can tell, spec-generator can process that document just fine, see https://labs.w3.org/spec-generator/?type=respec&url=https%3A%2F%2Fraw.githubusercontent.com%2Fw3c%2Fvc-jose-cose%2Fdcfe55993ade74facbd7d39d755e3a89dacefd66%2Findex.html&file=
@tobie, any idea why pr-preview isn't adding the links on that PR? The other PRs seem to work ok.

**Comment by iherman** [comment]
Hm. Maybe it is a problem with htmlpreview (I must admit I have never used this raw.githubusercontent.com trick).

File: index.html [hunk 1]
```diff
@@ -152,7 +152,7 @@
     </p>
 </section>
 
-<section id="introduction">
+<section class="informative" id="introduction">
```
Comment (TallTed) [inline index.html:155]: Non-normative/informative sections cannot contain normative subsections.

Therefore, if, as I believe will be the case when PR-Preview does its thing or when this PR is merged, the Terminology section, which is now meant to be normative, is a sub-section of the Introduction section, then the Introduction section *must* be made normative.
```suggestion
<section class="normative" id="introduction">
```

**Review by selfissued — CHANGES_REQUESTED** [review]
Please delete all instances of class="normative".  I verified that it doesn't change the output for normative sections, so it's just unnecessary clutter in the source, given that sections are normative by default.

**Comment by tobie** [comment]
> @tobie, any idea why pr-preview isn't adding the links on that PR? The other PRs seem to work ok.

Not sure what the issue was here. Seems fixed now. LMK if this happens again.

File: index.html [hunk 2]
```diff
@@ -147,7 +147,7 @@
     </p>
 </section>
 
-<section id="introduction">
+<section class="normative" id="introduction">
```
Comment (decentralgabe) [inline index.html:150]:
```suggestion
<section id="introduction">
```

File: index.html [hunk 3]
```diff
@@ -299,7 +299,7 @@ <h3 id="terminology">Terminology</h3>
     <div data-include="./terms.html"></div>
 </section>
 
-<section>
+<section class="normative">
```
Comment (decentralgabe) [inline index.html:302]:
```suggestion
<section>
```

File: index.html [hunk 4]
```diff
@@ -334,9 +334,9 @@ <h2 id="securing-the-vc-data-model">Securing the VC Data Model</h2>
         If implementations do not know which media type to use, media types defined in this specification MUST be used.
     </p>
 
-    <section id="secure-with-jose">
+    <section class="normative" id="secure-with-jose">
```
Comment (decentralgabe) [inline index.html:337]:
```suggestion
    <section id="secure-with-jose">
```

File: index.html [hunk 5]
```diff
@@ -398,7 +398,7 @@ <h2 id="securing-with-jose">Securing JSON-LD Verifiable Credentials with JOSE</h
             </p>
         </section>
 
-        <section>
+        <section class="normative">
```
Comment (decentralgabe) [inline index.html:401]:
```suggestion
        <section>
```

File: index.html [hunk 6]
```diff
@@ -557,9 +557,9 @@ <h2 id="jose-header-parameters-jwt-claims">JOSE Header Parameters and JWT Claims
         </section>
     </section>
 
-    <section id="secure-with-sd-jwt">
+    <section class="normative" id="secure-with-sd-jwt">
```
Comment (decentralgabe) [inline index.html:560]:
```suggestion
    <section id="secure-with-sd-jwt">
```

File: index.html [hunk 7]
```diff
@@ -633,7 +633,7 @@ <h2 id="securing-with-sd-jwt">Securing JSON-LD Verifiable Credentials with SD-JW
             </p>
         </section>
 
-        <section>
+        <section class="normative">
```
Comment (decentralgabe) [inline index.html:636]:
```suggestion
        <section>
```

File: index.html [hunk 8]
```diff
@@ -731,7 +731,7 @@ <h2 id="securing-vps-sd-jwt">Securing JSON-LD Verifiable Presentations with SD-J
         </section>
     </section>
 
-    <section id="secure-with-cose">
+    <section class="normative" id="secure-with-cose">
```
Comment (decentralgabe) [inline index.html:734]:
```suggestion
    <section id="secure-with-cose">
```

File: index.html [hunk 9]
```diff
@@ -740,7 +740,7 @@ <h2 id="securing-with-cose">With COSE</h2>
             use of content types as outlined in this section.
         </p>
 
-        <section>
+        <section class="normative">
```
Comment (decentralgabe) [inline index.html:743]:
```suggestion
        <section>
```

File: index.html [hunk 10]
```diff
@@ -803,7 +803,7 @@ <h2 id="securing-vcs-with-cose">Securing JSON-LD Verifiable Credentials with COS
                 for more details regarding this example.
             </p>
         </section>
-        <section>
+        <section class="normative">
```
Comment (decentralgabe) [inline index.html:806]:
```suggestion
        <section>
```

File: index.html [hunk 11]
```diff
@@ -953,13 +953,13 @@ <h2 id="key-discovery">Key Discovery</h2>
     </p>
 
 
-    <section>
+    <section class="normative">
```
Comment (decentralgabe) [inline index.html:956]:
```suggestion
    <section>
```

File: index.html [hunk 12]
```diff
@@ -953,13 +953,13 @@ <h2 id="key-discovery">Key Discovery</h2>
...
         <h2 id="using-header-params-claims-key-discovery">Using Header Parameters and Claims for Key Discovery</h2>
         <p>
             These JOSE header parameters and JWT claims can be used by
             <a data-cite="VC-DATA-MODEL-2.0#dfn-verifier">verifiers</a> to discover verification keys.
         </p>
-        <section>
+        <section class="normative">
```
Comment (decentralgabe) [inline index.html:962]:
```suggestion
        <section>
```

File: index.html [hunk 13]
```diff
@@ -975,7 +975,7 @@ <h2 id="kid">kid</h2>
                 or <a data-cite="VC-DATA-MODEL-2.0#dfn-subjects">subject</a> is expressed as a DID URL.
             </p>
         </section>
-        <section>
+        <section class="normative">
```
Comment (decentralgabe) [inline index.html:978]:
```suggestion
        <section>
```

File: index.html [hunk 14]
```diff
@@ -1000,7 +1000,7 @@ <h2 id="iss">iss</h2>
             </p>
         </section>
 
-        <section>
+        <section class="normative">
```
Comment (decentralgabe) [inline index.html:1003]:
```suggestion
        <section>
```

File: index.html [hunk 15]
```diff
@@ -1019,8 +1019,11 @@ <h2 id="cnf">cnf</h2>
         </section>
     </section>
 
-    <section>
+    <section class="normative">
         <h2 id="well-known-uris">Well-Known URIs</h2>
+
+        <section class="normative">
```
Comment (decentralgabe) [inline index.html:1025]:
```suggestion
        <section>
```

File: index.html [hunk 16]
```diff
@@ -1039,9 +1042,10 @@ <h2 id="well-known-uris">Well-Known URIs</h2>
           "kid": "https://vendor.example/issuers/42/keys/urn:ietf:params:oauth:jwk-thumbprint:sha-256:NzbLsXh8uDCcd-6MNwXF4W_7noWXFZAfHkxZsRGC9Xs"
         }
         </pre>
+        </section>
     </section>
 
-    <section>
+    <section class="normative">
```
Comment (decentralgabe) [inline index.html:1048]:
```suggestion
    <section class="normative">
```

File: index.html [hunk 17]
```diff
@@ -1152,6 +1156,254 @@ <h3 id="using-did-documents">Using DID Documents</h3>
 </section>
 
 <section class="normative">
```
Comment (decentralgabe) [inline index.html:1158]:
```suggestion
<section>
```

File: index.html [hunk 18]
```diff
@@ -1152,6 +1156,254 @@ <h3 id="using-did-documents">Using DID Documents</h3>
...
+            <code>true</code> for success and <code>false</code> for failure.
+        </li>
+        <li>
+            <code>document</code>: a document conforming to the [[[VC-DATA-MODEL-2.0]]] [[VC-DATA-MODEL-2.0]]
+        </li>
+        <li>
+            <code>mediaType</code>: <code>vc</code> or <code>vp</code>
+        </li>
+    </ul>
+    <section class="normative">
```
Comment (decentralgabe) [inline index.html:1189]:
```suggestion
    <section>
```

File: index.html [hunk 19]
```diff
@@ -1152,6 +1156,254 @@ <h3 id="using-did-documents">Using DID Documents</h3>
...
+                        Set <code>mediaType</code> to <code>null</code>
+                    </li>
+                    <li>
+                        Return
+                    </li>
+                </ol>
+            </li>
+        </ol>
+    </section>
+    <section class="normative">
```
Comment (decentralgabe) [inline index.html:1247]:
```suggestion
    <section>
```

**Review by selfissued — APPROVED** [review]

**Comment by decentralgabe** [comment]
Opened over a week. Changes addressed. Editorial approval. Merging.

**Pull request merged**